subsystem/cluster_test.go

```diff
@@ -909,9 +909,9 @@ var _ = Describe("Validate BaseDNSDomain when creating a cluster", func() {
 	}
 	tests := []DNSTest{
 		{
-			It:            "V2RegisterCluster should throw an error. BaseDNSDomain='example', not a valid DNS structure string",
+			It:            "V2RegisterCluster should not throw an error. BaseDNSDomain='example', valid DNS",
 			BaseDNSDomain: "example",
-			ShouldThrow:   true,
+			ShouldThrow:   false,
```

Comment:
Can you please add a subsystem test with a negative scenario? I.e. it's good to have `domain=example` that will succeed, but I'd also like to see a subsystem test with some domain that will fail, e.g. `basednsdomain=a-.com` and `shouldthrow: true`.

Apart from that looks good, this one test and LGTM

Reply:
This is a classic example for a code that does not need subsystem at all. The algorithm is tested thoroughly with unit test (as it should be) and there is no need for extra subsystem test that just add a thin layer of calling the algorithm through REST.

Reply:
Plus I missed the part of the subsystem but of course fixed it already. "example" should be a legal domain now

Reply:
Ok, fair enough 👍🏻